mobile/src/main/java/com/truckmuncher/truckmuncher/customer/CustomerMapFragment.java

```diff
@@ -237,10 +249,50 @@ public boolean onClusterClick(Cluster<TruckCluster> cluster) {
         return true;
     }
 
-    public void loadActiveTrucks(String searchQuery) {
+    public void searchTrucks(final String query) {
+        if (query == null || query.isEmpty()) {
+            forceClusterRender(activeTruckMarkers.values());
+        } else {
+            Thread t = new Thread(new Runnable() {
```

Comment:
This should be done in an `IntentService`. This approach is likely to lead to issues with rotation